templates/web/pages/product_edit/product_edit_form_display.tt.js

```diff
@@ -34,3 +34,67 @@
 		\$('#nutrition_data_div').show();
 	}
 });
+
+function show_warning(should_show, nutrient_id, warning_message){
+	if(should_show) {
+		\$('#nutriment_'+nutrient_id).css("background-color", "rgb(255 237 235)");
+		\$('#nutriment_question_mark_'+nutrient_id).css("display", "inline-table");
+		\$('#nutriment_sugars_warning_'+nutrient_id).text(warning_message);
+	}else {
+		\$('#nutriment_'+nutrient_id).css("background-color", "white");
+		\$('#nutriment_question_mark_'+nutrient_id).css("display", "none");
+	}
+}
+
+var sugars_value;
+var carbohydrates_value;
+var saturated_fats_value;
+var fat_value;
+
+var required_nutrients_id = ['energy-kj', 'energy-kcal', 'fat', 'saturated-fat', 'sugars', 'carbohydrates', 'fiber', 'proteins', 'salt', 'sodium', 'alcohol'];
+
+required_nutrients_id.forEach(nutrient_id => {
+	\$('#nutriment_' + nutrient_id).on('input', function() {
+		var nutrient_value = \$(this).val();
+		var is_above_or_below_100 = isNaN(nutrient_value) || nutrient_value < 0 || nutrient_value > 100;
+		show_warning(is_above_or_below_100, nutrient_id, "Please enter a value between 0 and 100");
```

Comment:
The warnings should be added to po/common/common.pot and po/common/en.po so they can be translated, but we can also do that later in another PR.

Reply:
@stephanegigandet can we do the translations in another PR
i have tried them locally but it might take me some time to make the logic work.